### ISSUE TITLE
TranslateBehavior incompatible with multiple default locales

### ISSUE THREAD

**Issue by SaronGrave** [cakephp/cakephp#13610]
This is a (multiple allowed):

* [ ] bug
* [ ] enhancement
* [x] feature-discussion (RFC)

* CakePHP Version: 3.8.3

### What happened
We have multiple instances of the same CakePHP software running live. This because of reason I won't go in to now. Some of these different instances do not have the same APP_DEFAULT_LOCALE set, because the users are from a different country/language.
We were trying to use CakePHPs `Cake\ORM\Behavior\TranslateBehavior.` This worked all fine until we tried it on instance with a different APP_DEFAULT_LOCALE. There the default used translation would be wrong. TranslateBehavior seems to expect that the 'default' field value is in the same locale as APP_DEFAULT_LOCALE. Setting `i18n::setLocale()` to the firts-used default does not seem to help, because there is no translation for it (because it is the original value in the table). 

### So in conclusion:
It is currently impossible to use the same database seed with different APP_DEFAULT_LOCALEs if you use `Cake\ORM\Behavior\TranslateBehavior`.

Conditional seeding based on locales could be done, but not for existing databases. Converting/replacing scripts would be needed. Seems unnecessarily much work.

Something like being able to overwrite TranslateBehaviors use of APP_DEFAULT_LOCALE seems like a solution...

What do you think?

**Comment by markstory** [comment]
I don't think I understand this scenario well enough to have an opinion. What would an override to the default do other than act as a default?

**Comment by SaronGrave** [comment]
Ok, I will try to explain the scenario some more:

So we two instances of the same software running.
Instance 1 has `'defaultLocale' => env('APP_DEFAULT_LOCALE', 'nl_NL'),` in app.php
Instance 2 has `'defaultLocale' => env('APP_DEFAULT_LOCALE', 'en_US'),` in app.php
Everything else (except the database connection) is exactly the same.

We have one table we are trying to use `Cake\ORM\Behavior\TranslateBehavior` in.

The `patient_types` table which looks like this:
![image](https://user-images.githubusercontent.com/2298583/64331722-9d8fb200-cfd3-11e9-8b6d-11d80a9b7438.png)

I added translations to the i18n table:
![image](https://user-images.githubusercontent.com/2298583/64331714-9a94c180-cfd3-11e9-90e1-c5484c5b0dad.png)

The PatientTypesTable has `$this->addBehavior('Translate', ['fields' => ['title', 'short_description', 'full_description']]);`
The PatientType entity has `use TranslateTrait;`

Testcode:
```php
I18n::setLocale('nl_NL');
$this->loadModel('PatientTypes');
$type = $this->PatientTypes->get(1);
echo $type->title;
I18n::setLocale('en_US');
$type = $this->PatientTypes->get(1);
echo $type->title;
```

On instance 1 the result is:
"Elderly care intramural" -> en_US translation
"Verzorgingstehuis intramuraal" -> nl_NL translation
Works as expected.

But on instance 2 the result is:
"Elderly care intramural" -> default database value (in en_US)
"Elderly care intramural" -> en_US translation

Editing the data in the DB shows that when the APP_DEFAULT_LOCALE and I18n::setLocale() are the same, TranslateBehavior gives the default value from the patient_types table. Not the translation that is available.
But this makes it impossible to use different APP_DEFAULT_LOCALEs with the same database (copy).
I cannot find a way to override this behavior. 
As I want TranslateBehavior to first check if there is a matching entry in the i18n table, despite it maybe is in the same locale as APP_DEFAULT_LOCALE.

**Comment by garas** [comment]
As your main table always contain 'en_US' and not always matches `App.defaultLocale`, add option `'defaultLocale' => 'en_US'`  to `$this->addBehavior()`.

**Comment by SaronGrave** [comment]
Yes, thanks. That worked. Maybe this should have some place in the docs? Or did I miss it somewhere?

**Comment by ADmad** [comment]
@SaronGrave You can submit an update to the docs by clicking the pencil icon on the right on https://book.cakephp.org/3.0/en/orm/behaviors/translate.html
